### PR TITLE
add initial support for audio type

### PR DESCRIPTION
**Pull request by sandalns**
### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

**Comment by sandalns** [comment]
Open questions that I'm looking at:
- Are there any additional Annotation types that we should add for audio types?
- How would GroundTruths and Inferences potentially look like for for audio data?

**Comment by dylangrandmont** [comment]
We might as well make sure `audio` is supported for datapoints as well

**Comment by ly-kolena** [comment]
> We might as well make sure `audio` is supported for datapoints as well

Why is this highlighted? annotations/assets are not specific to datapoints/testsample.

**Comment by dylangrandmont** [comment]
> Why is this highlighted? annotations/assets are not specific to datapoints/testsample.

Oh, this is my bad. I thought we were going to support audio as a new test sample `data_type` in which case we would want to support it as a datapoint `data_type` as well. Which raises the question... are we going to be supporting this as a test sample type?

**Comment by sandalns** [comment]
> > Why is this highlighted? annotations/assets are not specific to datapoints/testsample.
> 
> Oh, this is my bad. I thought we were going to support audio as a new test sample `data_type` in which case we would want to support it as a datapoint `data_type` as well. Which raises the question... are we going to be supporting this as a test sample type?

I am adding audio as a test sample type, just initially only added the asset type. I will add it to datapoint data_type after https://github.com/kolenaIO/kolena/pull/258 is merged for simplicity.

File: kolena/workflow/test_sample.py [hunk 1]
```diff
@@ -286,7 +287,19 @@ def _data_type(cls) -> _TestSampleType:
         return _TestSampleType.POINT_CLOUD
 
 
-_TEST_SAMPLE_BASE_TYPES = [Composite, Image, Text, BaseVideo, Document, PointCloud]
+@dataclass(frozen=True, config=ValidatorConfig)
+class Audio(TestSample):
+    """An audio file located in a cloud bucket or served at a URL."""
+
+    locator: str
+    """URL (e.g. S3, HTTPS) of the audio file."""
```
Comment (gordonhart) [inline kolena/workflow/test_sample.py:295]: Are there any restrictions on the format (e.g. `.mp3`)?

Comment (sandalns) [reply kolena/workflow/test_sample.py:295]: Will update this with the supported file types.

File: kolena/workflow/__init__.py [hunk 2]
```diff
@@ -87,4 +88,5 @@
     "TestRun",
     "test",
     "define_workflow",
+    "Audio",
```
Comment (gordonhart) [inline kolena/workflow/__init__.py]: Should be ordered with the rest of the test sample types

File: kolena/workflow/test_sample.py [hunk 3]
```diff
@@ -113,6 +113,7 @@ class _TestSampleType(DataType):
     DOCUMENT = "DOCUMENT"
     COMPOSITE = "COMPOSITE"
     POINT_CLOUD = "POINT_CLOUD"
+    AUDIO = "AUDIO"
```
Comment (gordonhart) [inline kolena/workflow/test_sample.py:116]: Add to datapoint types as well?

**Review by gordonhart — APPROVED** [review]
Let's land this (after adding to datapoint types)!

File: kolena/workflow/test_sample.py (hunk 3 above)
Comment (sandalns) [reply kolena/workflow/test_sample.py:116]: done

File: kolena/workflow/__init__.py (hunk 2 above)
Comment (sandalns) [reply kolena/workflow/__init__.py]: done

**Pull request merged**